src/CryptoPaymentsSigner.js

```diff
@@ -0,0 +1,115 @@
+  constructor({ paymentsAddr, erc20Addr, eth }) {
+    this.erc20Contract = new eth.Contract(IERC20JSON.abi, erc20Addr);
+    this.paymentsContract = new eth.Contract(PaymentsJSON.abi, paymentsAddr);
+  }
+
+  async registerAsSeller({ from }) {
+    return this.paymentsContract.methods.registerAsSeller().send({ from });
+  }
+
+  async withdraw({ from }) {
+    return this.paymentsContract.methods.withdraw().send({ from });
+  }
+
+  async pay({ paymentData, signature, from }) {
+    return this.paymentsContract.methods.pay(paymentData, signature).send({ from });
+  }
+
+  async approve({ spender, amount, from }) {
+    return this.erc20Contract.methods.approve(spender, amount).send({ from });
+  }
+
+  async erc20BalanceOf({ address }) {
+    return this.paymentsContract.methods.erc20BalanceOf(address).call();
+  }
+
+  async balanceOf({ address }) {
+    return this.paymentsContract.methods.balanceOf(address).call();
+  }
+
+  async allowance({ address }) {
+    return this.paymentsContract.methods.allowance(address).call();
+  }
+
+  async isRegisteredSeller({ address }) {
+    return this.paymentsContract.methods.isRegisteredSeller(address).call();
+  }
+
+  async enoughFundsAvailable({ address, amount }) {
+    return this.paymentsContract.methods.enoughFundsAvailable(address, amount).call();
+  }
+
+  async maxFundsAvailable({ address }) {
+    return this.paymentsContract.methods.maxFundsAvailable(address).call();
+  }
+
+  async computeFeeAmount({ amount, feeBPS }) {
+    return this.paymentsContract.methods.computeFeeAmount(amount, feeBPS).call();
+  }
+
+  async splitFundingSources({ address, amount }) {
+    return this.paymentsContract.methods.splitFundingSources(address, amount).call();
+  }
+
+  async paymentState({ paymentId }) {
+    return this.paymentsContract.methods.paymentState(paymentId).call();
+  }
+
+  async paymentWindow() {
+    return this.paymentsContract.methods.paymentWindow().call();
+  }
+
+  async acceptedCurrency() {
+    return this.paymentsContract.methods.acceptedCurrency().call();
+  }
+
+  static isValidPaymentData({ paymentData }) {
+    if (!Object.prototype.hasOwnProperty.call(paymentData, 'paymentId')) return false;
+    if (!Object.prototype.hasOwnProperty.call(paymentData, 'amount')) return false;
+    if (!Object.prototype.hasOwnProperty.call(paymentData, 'feeBPS')) return false;
+    if (!Object.prototype.hasOwnProperty.call(paymentData, 'universeId')) return false;
+    if (!Object.prototype.hasOwnProperty.call(paymentData, 'validUntil')) return false;
+    if (!Object.prototype.hasOwnProperty.call(paymentData, 'buyer')) return false;
+    if (!Object.prototype.hasOwnProperty.call(paymentData, 'seller')) return false;
+    if (paymentData.feeBPS > 10000) return false;
+    if (paymentData.buyer === paymentData.seller) return false;
+    return true;
+  }
+
+  static PaymentStates() {
+    return {
+      NotStarted: 0,
+      AssetTransferring: 1,
+      Failed: 2,
+      Paid: 3,
+    };
+  }
```

Comment:
We should be able to change this to use class fields: https://javascript.info/class#class-fields

```suggestion
  paymentStates = {
      NotStarted: 0,
      AssetTransferring: 1,
      Failed: 2,
      Paid: 3
  };
```

Reply:
ESLINT complains if I do as suggested too, but with little info: 
![image](https://user-images.githubusercontent.com/43603913/150840816-970ad253-10ea-4722-8492-d6b0bb3ec76d.png)